transferui/src/main/java/com/hyperwallet/android/ui/transfer/view/ListTransferDestinationFragment.java

```diff
@@ -0,0 +1,204 @@
+                        mAdapter.replaceData(destinations);
+                    }
+                });
+    }
+
+    @Override
+    public void onViewStateRestored(@Nullable Bundle savedInstanceState) {
+        super.onViewStateRestored(savedInstanceState);
+
+        if (savedInstanceState != null) {
+            mSelectedDestination = savedInstanceState.getParcelable(ARGUMENT_SELECTED_TRANSFER_DESTINATION);
+        } else {
+            mSelectedDestination = getArguments().getParcelable(ARGUMENT_SELECTED_TRANSFER_DESTINATION);
+        }
+
+        mAdapter = new ListTransferDestinationAdapter(mSelectedDestination,
+                mSelectDestinationItemClickListener);
+        mRecyclerView.setAdapter(mAdapter);
+    }
+
+    @Override
+    public void onSaveInstanceState(@NonNull Bundle outState) {
+        outState.putParcelable(ARGUMENT_SELECTED_TRANSFER_DESTINATION, mSelectedDestination);
+        super.onSaveInstanceState(outState);
+    }
+
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                NavUtils.navigateUpFromSameTask(getActivity());
+                return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public void onClose() {
+        getActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
+        getActivity().getWindow().addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+        getActivity().getWindow().setStatusBarColor(ContextCompat.getColor(getContext(), R.color.colorPrimaryDark));
+        getActivity().getWindow().getDecorView().setSystemUiVisibility(0);
+    }
+
+    @Override
+    public void onView() {
+        getActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
+        getActivity().getWindow().addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            getActivity().getWindow().setStatusBarColor(
+                    ContextCompat.getColor(getContext(), R.color.regularColorPrimary));
+            getActivity().getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR);
+        }
+    }
+
+    @Override
+    public void hideSoftKey(@NonNull View focusedView) {
+        InputMethodManager inputMethodManager = (InputMethodManager) focusedView.getContext().getSystemService(
+                Activity.INPUT_METHOD_SERVICE);
+        inputMethodManager.hideSoftInputFromWindow(focusedView.getWindowToken(), 0);
+    }
+
+    @Override
+    public void selectTransferDestination(int position) {
+        HyperwalletTransferMethod destination = mAdapter.getItem(position);
+        mListTransferDestinationViewModel.getTransferDestinationSection().setValue(new Event<>(destination));
```

Comment:
I think its better to call `mListTransferDestinationViewModel.selectTransferDestination(destination);` rather than setting the value directly

Reply:
changed